File: src/features/ai-platform/infra/commands/agents.ts
import { invoke } from '@tauri-apps/api/core';

export interface AgentCategoryRecord {
  id: string;
  name: string;
  description: string;
}

export interface AgentAssignmentRecord {
  agentKey: string;
  providerId?: string;
  model?: string;
  timeoutSeconds?: number;
}

export interface AgentProviderOption {
  providerId: string;
  providerName: string;
  model: string;
}

export interface AgentSnapshotItem {
  key: string;
  name: string;
  category: string;
  description: string;
  enabled: boolean;
  assignment: AgentAssignmentRecord;
}

export interface AgentsSnapshot {
  categories: AgentCategoryRecord[];
  agents: AgentSnapshotItem[];
  providerOptions: AgentProviderOption[];
  routingMode: string;
  source: string;
}

export function getAiPlatformAgentsSnapshot() {
  return invoke<AgentsSnapshot>('ai_platform_get_agents_snapshot');
}

export function setAiPlatformAgentEnabled(agentKey: string, enabled: boolean) {
  return invoke<AgentsSnapshot>('ai_platform_set_agent_enabled', { agentKey, enabled });
}

export function saveAiPlatformAgentAssignment(assignment: AgentAssignmentRecord) {
  return invoke<AgentsSnapshot>('ai_platform_save_agent_assignment', { assignment });
}

export function setAiPlatformAgentsRoutingMode(routingMode: string) {
  return invoke<AgentsSnapshot>('ai_platform_set_agents_routing_mode', { routingMode });
}

File: src-tauri/src/ai_platform/interfaces/dto/agents.rs
use crate::ai_platform::domain::agent::{AgentAssignmentRecord, AgentCategoryRecord};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AgentProviderOptionDto {
    pub provider_id: String,
    pub provider_name: String,
    pub model: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AgentSnapshotItemDto {
    pub key: String,
    pub name: String,
    pub category: String,
    pub description: String,
    pub enabled: bool,
    pub assignment: AgentAssignmentRecord,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AgentsSnapshotDto {
    pub categories: Vec<AgentCategoryRecord>,
    pub agents: Vec<AgentSnapshotItemDto>,
    pub provider_options: Vec<AgentProviderOptionDto>,
    pub routing_mode: String,
    pub source: String,
}